### PR TITLE
cargo: rename crates from `jujutsu`/`jujutsu-lib` to `jj-cli`/`jj-lib`

### PR DESCRIPTION
**Pull request by martinvonz**
I plan to merge this soon before we release 0.8.0, which we should be able to do pretty soon, I think.

# Checklist

If applicable:
- [ ] I have updated `CHANGELOG.md`
- [ ] I have updated the documentation (README.md, docs/, demos/)
- [ ] I have updated the config schema (src/config-schema.json)
- [ ] I have added tests to cover my changes

### CODE REVIEW

**Comment by necauqua** [comment]
I'd add this commit to `.git-blame-ignore-revs` which github recognizes

**Comment by martinvonz** [comment]
> I'd add this commit to `.git-blame-ignore-revs` which github recognizes

I didn't know that GitHub had that feature. I guess we'll have to do that after this gets merged.

@necauqua: if you haven't already, could you take a look at the Nix flake and let me know which strings to update there?

**Comment by martinvonz** [comment]
> All instances of `jujutsu` string are the nix package/overlay/app names which we should keep as jujutsu imo (unless we rename it entirely); and also the "jj" is the name of the binary which we didn't rename :)
> 
> It just builds the cargo project in the current ~folder~ repo. So as far as I can tell we don't need to update anything for nix and it should just work

Thanks for checking. I didn't know if any of them were just local identifiers that we can rename as we like.

> You can just install nix (on linux/mac) without going all in with NixOS and run `nix build/run` to sanity-check stuff immediately :) Nix is awesome

Yes, I actually did that and it seemed to work (`nix --extra-experimental-features nix-command --extra-experimental-features flakes build` is what I used).

**Comment by necauqua** [comment]
yeah the experimental features existed long enough and were superior to legacy alternatives for long enough for it to be forgotten about, so `echo 'experimental-features = nix-command flakes' >> ~/.config/nix/nix.conf` :)

maybe hopefully soon they'll finally mark those as non experimental

File: Cargo.toml [hunk 1]
```diff
@@ -8,13 +8,13 @@ license = "Apache-2.0"
...
 repository = "https://github.com/martinvonz/jj"
-documentation = "https://docs.rs/jujutsu"
+documentation = "https://docs.rs/jj-cli"
 readme = "README.md"
 keywords = ["VCS", "DVCS", "SCM", "Git", "Mercurial"]
 categories = ["command-line-utilities", "development-tools"]
 default-run = "jj"
 
-[[bin]]
+[[bin]]
```
Comment (ilyagr) [inline Cargo.toml:17]: There's an errant space at the end of the line.

Comment (martinvonz) [reply Cargo.toml]: Fixed

**Review by ilyagr — APPROVED** [review]
I looked a bit, and couldn't find anything you missed. I didn't check Nix-related things, but I think you and @necauqua already double-checked that.

**Pull request merged**